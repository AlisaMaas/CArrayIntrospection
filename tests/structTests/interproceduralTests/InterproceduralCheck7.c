#include "structHeader.h"
/**
 * This tests that the case in which the same formal argument
 * is used as multiple arguments. It should find that string is non-null terminated.
 **/
void print(char *string) {}
int find(struct foo *f) {
	print(f->nonString);
	for (int i = 0;; i++) {
		if (f->nonString[i] != '\0') {
			print("About to break from loop");
			break;
		}
	}
	for (int i = 0;; i++) {
		if (f->string[i] == '\0') {
			print("About to break from loop");
			break;
		}
	}
	return 1;
}
int foo(struct foo *f) {
	print(f->nonString);
	for (int i = 0;; i++) {
		if (f->nonString[i] != '\0') {
			print("About to break from loop");
			break;
		}
	}
	return find(f);
}
