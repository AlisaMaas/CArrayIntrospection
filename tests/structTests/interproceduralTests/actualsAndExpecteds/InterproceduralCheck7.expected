Printing analysis 'Promote Memory to Register' for function 'print':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'find':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'foo':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Determine whether and how to annotate each function with the null-terminated annotation':
struct struct.foo[0] should be annotated NULL_TERMINATED (2).
